### PR TITLE
Bulk registration

### PR DESCRIPTION
**Pull request by janeenyamak1**
## How did you test this change?

- [ ] Test(s) added
- [ ] Manual testing (please provide screenshots/recordings)
- [ ] No testing (please provide an explanation)

### CODE REVIEW

**Comment by akphi** [comment]
@janeenyamak1 can you add a changeset, per instruction https://github.com/finos/legend-studio/blob/master/CONTRIBUTING.md#changeset .Thanks!

File: .changeset/tiny-parrots-five.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@finos/legend-graph': patch
+---
+
+generating changeset
```
Comment (MauricioUyaguari) [inline .changeset/tiny-parrots-five.md:5]: 1. everything else 
2. 
3. ---
'@finos/legend-application-studio': minor
---

Enable Global Bulk Service Registration

File: .changeset/tiny-parrots-two.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+'@finos/legend-application-query': patch
+'@finos/legend-query-builder': patch
+'@finos/legend-application': patch
+'@finos/legend-graph': patch
+---
+
+generating changeset
```
Comment (akphi) [inline .changeset/tiny-parrots-two.md:8]:
```suggestion
```

**Review by akphi — APPROVED** [review]

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+export const RegisterService = observer(
+  (props: { bulkServiceRegistrationState: BulkServiceRegistrationState }) => {
+    const editorStore = useEditorStore();
+    const services = editorStore.graphManagerState.graph.ownServices;
+    const [open, setOpen] = useState(false);
+
+    const [isMaximized, setIsMaximized] = useState(false);
+    const toggleMaximize = (): void => setIsMaximized(!isMaximized);
+
+    const renderTestables = (): React.ReactNode => (
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:45]: lets change this to the correct name. `ServiceItem` and lets also not reuse the class names from testable but create separate css files for the buil service registration

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+            tabIndex={-1}
+            title="Register All Services"
+          >
+            <PlayIcon />
+          </button>
+          <Dialog onClose={noop} open={open}>
+            <Modal
+              darkMode={true}
+              className={clsx('editor-modal query-builder__dialog', {
+                'query-builder__dialog--expanded': isMaximized,
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:92]: lets create separate css classes and not reuse query builder ones

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+              })}
+            >
+              <div className="query-builder__dialog__header">
+                <div className="query-builder__dialog__header__actions"></div>
+                <button
+                  className="query-builder__dialog__header__action"
+                  tabIndex={-1}
+                  onClick={toggleMaximize}
+                >
+                  {isMaximized ? (
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:102]: Do we use `isMaximized` ?

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+} from '@finos/legend-art';
+import type { BulkServiceRegistrationState } from '../../../stores/sidebar-state/BulkServiceRegistrationState.js';
+import { BulkServiceRegistrationEditor } from '../edit-panel/service-editor/BulkServiceRegistrationEditor.js';
+import { noop } from '@finos/legend-shared';
+
+export const RegisterService = observer(
+  (props: { bulkServiceRegistrationState: BulkServiceRegistrationState }) => {
+    const editorStore = useEditorStore();
+    const services = editorStore.graphManagerState.graph.ownServices;
+    const [open, setOpen] = useState(false);
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:40]: lets rename open to be more descriptive `open` -> `showRegistrationModal` ?

File: packages/legend-application-studio/src/stores/sidebar-state/BulkServiceRegistrationState.ts [hunk 7]
```diff
@@ -0,0 +1,370 @@
...
+interface ServiceVersionOption {
+  label: string;
+  value: Version | string;
+}
+interface ServiceRegistrationResult {
+  successfulServices: string[];
+  failedServices: string[];
+  serviceLinks: string[];
+}
+export class ServiceConfigState {
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/stores/sidebar-state/BulkServiceRegistrationState.ts:74]: Can we think of a way to organize this model in conjuntion with `ServiceRegistrationState`. 
we may need one state to hold the config and both the serice element editor use it and the build registration one use it as well. 
something like maybe 
`ServiceRegistrationConfigState`
`BuildServiceRegistrationConfigState extends ServiceRegistrationConfigState`
`ServiceEditorServiceRegistrationConfigState extends ServiceRegistrationConfigState`
We want to minimize the repeated code in both states and think about what we share that is only relevant to the registration config

File: packages/legend-application-studio/src/components/editor/edit-panel/service-editor/BulkServiceRegistrationEditor.tsx [hunk 8]
```diff
@@ -0,0 +1,283 @@
...
+                  <>
+                    {editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.successfulServices.forEach(
+                      (element) => {
+                        <div>{element}</div>;
+                      },
+                    )}
+                  </>
+                  <div>
+                    Failed Services:{' '}
+                    {editorStore.bulkServiceRegistrationState.serviceConfigState.registrationResult?.failedServices.join(
```
Comment (MauricioUyaguari) [inline packages/legend-application-studio/src/components/editor/edit-panel/service-editor/BulkServiceRegistrationEditor.tsx:185]: Don't render lists like this. use a list of dom elements, like so:
`items.map(e => <div></div>)`

**Review by MauricioUyaguari — COMMENTED** [review]
Lets also make sure we update the PR with videos of the current buld registration flow

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx (hunk 4 above)
Comment (janeenyamak1) [reply packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:92]: created a bulk-service-registration.scss file

File: packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx (hunk 5 above)
Comment (janeenyamak1) [reply packages/legend-application-studio/src/components/editor/side-bar/RegisterService.tsx:102]: We do use it

**Pull request merged**